File: frontend/src/types/ticket.ts
export interface Ticket {
  id: number;
  title: string;
  description: string;
  status: number; // 0: Pending, 1: InProgress, 2: Done
  createdAt: string;
  updatedAt: string;
  closedAt?: string;
  contractId: number;
  houseName?: string;
  roomName?: string;
  tenantName?: string;
}

export interface CreateTicketDto {
  title: string;
  description: string;
}

export interface UpdateTicketStatusDto {
  status: string;
}

File: TicketService/Models/Ticket.cs
using TicketService.Models.Enums;

namespace TicketService.Models;

public class Ticket
{
    public int Id { get; set; }

    public string? TenantId { get; set; } 

    public string? OwnerId { get; set; }

    public int ContractId { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public TicketStatus Status { get; set; } = TicketStatus.Pending;

    public DateTime CreatedAt { get; set; } = DateTime.Now;

    public DateTime UpdatedAt { get; set; } = DateTime.Now;

    public DateTime ClosedAt { get; set; }
}